docs/user-guide/deploy-on-OSP.md

```diff
@@ -1,2 +1,68 @@
-# Openshift deployment with OAS - On Openstack
-work in progress...
+# Openshift Deployment on OpenStack by OpenShift Assisted Service 
+
+This guide explains how to deploy OpenShift by the OpenShift Assisted Service
+on OpenStack.
+
+---
+**NOTE**
+
+Currently, the deployment of an OpenShift Cluster on
+Red Hat OpenStack Platform is blocked by OpenShift Assisted Service, while RDO is working fine.
+The related check could be disabled by including `valid-platform` the environment variable
+`DISABLED_HOST_VALIDATIONS` in the conext of the OpenShift Assisted Service, e.g. like this:
+```
+DISABLED_HOST_VALIDATIONS=valid-platform,container-images-available 
+```
+
+---
+
+## Requirements
+
+* Two floating IP addresses, it is recommended that DNS resolves
+  * api.CLUSTERNAME.DOMAIN to the first floating IP address, and
+  * oauth-openshift.apps.CLUSTERNAME.DOMAIN and
+    console-openshift-console.apps.CLUSTERNAME.DOMAIN to the second one
+* The resources required by the VMs
+* A dedicated OpenStack network and subnet are recommended
+
+## Steps
+
+
+1. Generate the discovery iso in OpenShift Assisted Service.
+2. Upload the discovery iso as a new image to OpenStack.
+3. Create two ports and associate each one a floating IP address.
+4. Create the VMs to run the OpenShift cluster, with a bootable fresh volume
+   and the image of the discovery iso as second boot index. Please find an example below.
+5. Add the addresses of both floating IPs as the `allowed_address_pairs` of all VMs which
+   might run a master role in the OpenShift cluster. This will enable the virtual IPs to be
+   usable on the VMs. Please the details in  
+   [Highly available VIPs on OpenStack VMs with VRRP](https://blog.codecentric.de/en/2016/11/highly-available-vips-openstack-vms-vrrp/)
```

Comment:
does this article still apply? the article is from 2016..
Also, why not use Octavia as an OpenStack native loadbalancer to manage that VIP? https://github.com/openstack/octavia

Reply:
> does this article still apply? the article is from 2016..

@otherwiseguy Are you aware of any better documentation of the usage of virtual IPs in Neutron?

Reply:
> Also, why not use Octavia as an OpenStack native loadbalancer to manage that VIP? https://github.com/openstack/octavia

In the first place, because this is not supported by assisted installer. But never the less, it is an excellent question to understand if the assisted installer should utilize them.
@Fedosin is IPI installer able to utilize the OpenStack native loadbalancer?

Reply:
> > Also, why not use Octavia as an OpenStack native loadbalancer to manage that VIP? https://github.com/openstack/octavia
> 
> In the first place, because this is not supported by assisted installer. But never the less, it is an excellent question to understand if the assisted installer should utilize them.
> @Fedosin is IPI installer able to utilize the OpenStack native loadbalancer?

No. This is something we're looking at for future releases but this isn't supported for now.

Reply:
> does this article still apply? the article is from 2016..
> Also, why not use Octavia as an OpenStack native loadbalancer to manage that VIP? https://github.com/openstack/octavia

@umago Are you aware of any better documentation of the usage of virtual IPs in Neutron?

Reply:
Hi dominik,

Not in OpenStack itself, perhaps it would be good to add one. But, I remember that I used this article as reference for implementing virtual ports in ML2/OVN: https://docs.catalystcloud.nz/tutorials/compute/deploying-highly-available-instances-with-keepalived.html#deploying-highly-available-instances-with-keepalived

The way it works is: Upon adding an IP address to the "allowed_address_pairs" field in the Neutron's port the ML2/OVN driver will check if that IP matches with the IP of another existing port in the same network (Logical_Switch in OVN) and, if they do match, ML2/OVN will update the type of the matching port to "virtual". That's basically how we use OVN's virtual ports feature in OpenStack.

Reply:
Hi Lucas,

> Not in OpenStack itself, perhaps it would be good to add one. But, I remember that I used this article as reference for implementing virtual ports in ML2/OVN: https://docs.catalystcloud.nz/tutorials/compute/deploying-highly-available-instances-with-keepalived.html#deploying-highly-available-instances-with-keepalived

I added this is a nice reference, too.

> The way it works is: Upon adding an IP address to the "allowed_address_pairs" field in the Neutron's port the ML2/OVN driver will check if that IP matches with the IP of another existing port in the same network (Logical_Switch in OVN) and, if they do match, ML2/OVN will update the type of the matching port to "virtual". That's basically how we use OVN's virtual ports feature in OpenStack.

I added your tight explanation to the text. Even there is no official documentation, the user will have a rough understanding what is going on.